webpack/assets/javascripts/react_app/common/Tour/TourActions.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable promise/prefer-await-to-then */
+import API from '../../API';
+import {
+  TOUR_UPDATE_STATUS,
+  TOUR_GET_STATUSES,
+  TOUR_START_RUNNIG,
+  TOUR_STOP_RUNNING,
+  TOUR_REGISTER,
+  TOURS_REQUESTS,
+} from './TourConstants';
+
+export const updateAsSeen = id => async dispatch =>
+  API.post('/user_preferences', {
+    kind: 'Tour',
+    value: { alreadySeen: true },
+    name: id,
+  }).then(() =>
+    dispatch({
+      type: TOUR_UPDATE_STATUS,
+      payload: { id },
```

Comment:
This dispatch should happen even if the API call fails.   So maybe put it in a `finally` block (or don't tie it to the promise at all).